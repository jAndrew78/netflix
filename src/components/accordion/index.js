import React, { useState } from 'react';


export default function Accordion({ children, ...restProps }) {
    <Container {...restProps}>
        <Inner>{children}</Inner>
    </Container>
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
    return <Title {...restProps}>{ children }</Title>;
}

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
    return <Frame {...restProps}>{ children }</Frame>;
}

Accordion.Item = function AccordionItem({ children, ...restProps }) {
    return <Item {...restProps}>{ children }</Item>;
}

Accordion.Body = function AccordionBody({ children, ...restProps }) {
    return <Body {...restProps}>{ children }</Body>;
}

