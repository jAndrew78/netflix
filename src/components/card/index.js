import React from 'react';
import { Container, Group, Title, SubTitle, Text, } from './styles/card';


export default function Card({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>;
}

Card.Group = function CardGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}

Card.Title = function CardTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
}

Card.SubTitle = function CardSubTitle({ children, ...restProps }) {
    return <SubTitle {...restProps}>{children}</SubTitle>
}

Card.Text = function CardText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}
