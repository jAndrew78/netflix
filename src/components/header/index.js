import React, { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';

import { 
    Background, 
    Container, 
    Group, 
    Feature, 
    Logo, 
    ButtonLink, 
    Picture, 
    Profile,
    Dropdown,
    DropdownBreak,
    Break, 
    Text, 
    FeatureCallOut, 
    Link, 
    Search,
    SearchIcon,
    CloseIcon,
    SearchInput,
} from './styles/header';


export default function Header({ bg = true, children, ...restProps }) {
    return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
    return <Feature {...restProps}>{children}</Feature>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReactRouterLink to={to}>
            <Logo {...restProps} />
        </ReactRouterLink>
    )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
    const [searchActive, setSearchActive] = useState(false);

    return (
        <Search {...restProps}>
            <SearchIcon onClick={() => setSearchActive(searchActive => !searchActive)} active={searchActive}>
                <img src="/images/icons/search.png" alt="Search" />
            </SearchIcon>
            <SearchInput
                value={searchTerm}
                onChange={({ target }) => setSearchTerm(target.value)}
                placeholder="Search films and series"
                active={searchActive}
            />
            <CloseIcon onClick={() => setSearchActive(searchActive => !searchActive)} active={searchActive}>
                <img src="/images/icons/close-slim.png" alt="Close" />
            </CloseIcon>
        </Search>
    ) 
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.DropdownBreak = function HeaderDropdownBreak({ children, ...restProps }) {
    return <DropdownBreak {...restProps}>{children}</DropdownBreak>
}

Header.Break = function HeaderBreak({ children, ...restProps }) {
    return <Break {...restProps}>{children}</Break>
}

Header.Text = function HeaderText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>
}

