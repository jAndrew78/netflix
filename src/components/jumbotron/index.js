import React from 'react';
import { Inner, Container } from './styles/jumbotron';


export default function Jumbotron({ children, direction = 'row', ...restProps }) {    // get children, pass direction, then allowing any other props to be passed after
    return(
        <Inner direction={direction}>
            <p>Hello Jumbo</p>
        </Inner>
    );
}

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
    return <Container { ...restProps }>{children}</Container>;
}
