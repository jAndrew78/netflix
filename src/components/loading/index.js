import React from 'react';
import { Container } from './styles/loading';


export default function Loading({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}