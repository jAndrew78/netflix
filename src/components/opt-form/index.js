import React from 'react';
import { Container, Input, Button, Text } from './styles/opt-form';


export default function OptForm({ children, ...restProps }) {    // get children, then allow any other props to be passed after
    return <Container {...restProps}>{ children }</Container>;
}

OptForm.Input = function OptFormInput({ ...restProps }) {
    return <Input {...restProps} />;
}

OptForm.Button = function OptFormButton({ children, ...restProps }) {
    return <Button {...restProps}>{ children }</Button>;
}

OptForm.Text = function OptFormText({ children, ...restProps }) {
    return <Text {...restProps}>{ children }</Text>;
}

