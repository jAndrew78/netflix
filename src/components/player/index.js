import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';

import { Container, Button, Overlay, Inner, Close } from './styles/player';


// have to use context with styled components to be able to pass state around
export const PlayerContext = createContext();

export default function Player ({ children, ...restProps }) {
    const [showPlayer, setShowPlayer] = useState(false);

    return (
        <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
            <Container {...restProps}>{children}</Container>
        </PlayerContext.Provider>
    ) 
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
    const { showPlayer, setShowPlayer} = useContext(PlayerContext);

    return showPlayer ? ReactDOM.createPortal(
        <Overlay {...restProps}> 
            <Inner>
                <video id="netflix-player" controls>
                    <source src={src} type="video/mp4" />
                </video>
                <Close onClick={() => setShowPlayer(false)}> 
                    <img src="/images/icons/close.png" alt="Close" />
                </Close>
            </Inner>
        </Overlay>,
        document.body
    ) : null;
}

Player.Button = function PlayerButton({ ...restProps }) {
    const { showPlayer, setShowPlayer } = useContext(PlayerContext);

    return (
        <Button 
            onClick={() => setShowPlayer(showPlayer => !showPlayer)}>
            {/* use below to remove showPlayer error above, but may cause an error in rare cases */}
            {/* onClick={() => setShowPlayer(!showPlayer)}> */}
            <span>&#9654; </span>
            Play
        </Button>
    );
}
