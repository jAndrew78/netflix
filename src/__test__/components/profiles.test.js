import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../components';

describe('<Profiles />', () => {
    it('renders the <Profiles /> with populated data', () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Who's Watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}}>
                        <Profiles.Picture src='/images/andrew.png' data-testid="profile-picture" />
                        <Profiles.Name>Andrew</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>         
        );

        expect(getByText("Who's Watching?")).toBeTruthy();
        expect(getByTestId('profile-picture')).toBeTruthy();
        expect(getByText("Andrew")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the <Profiles /> with populated data but misc profile img', () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Who's Watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}}>
                        <Profiles.Picture src='/images/andrew.png' data-testid="profile-picture-misc" />
                        <Profiles.Name>Andrew</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>         
        );

        expect(getByText("Who's Watching?")).toBeTruthy();
        expect(getByTestId('profile-picture-misc')).toBeTruthy();
        expect(getByText("Andrew")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});