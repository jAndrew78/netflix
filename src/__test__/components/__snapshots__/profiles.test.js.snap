// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 fGoDNJ"
>
  <h1
    class="profiles__Title-bxtkkv-1 fMhCQW"
  >
    Who's Watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 dokCjq"
  >
    <li
      class="profiles__Item-bxtkkv-5 cDdBfI"
    >
      <img
        class="profiles__Picture-bxtkkv-3 fBRVOq"
        data-testid="profile-picture"
        src="images/users//images/andrew.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-4 imwOXw"
      >
        Andrew
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders the <Profiles /> with populated data but misc profile img 1`] = `
<div
  class="profiles__Container-bxtkkv-0 fGoDNJ"
>
  <h1
    class="profiles__Title-bxtkkv-1 fMhCQW"
  >
    Who's Watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 dokCjq"
  >
    <li
      class="profiles__Item-bxtkkv-5 cDdBfI"
    >
      <img
        class="profiles__Picture-bxtkkv-3 fBRVOq"
        data-testid="profile-picture-misc"
        src="images/users//images/andrew.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-4 imwOXw"
      >
        Andrew
      </p>
    </li>
  </ul>
</div>
`;
